### PR TITLE
refactor: players repository (& unrelated refactoring that really shouldn't be in this pr)

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: app/repositories/players.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+async def create(
+    name: str,
+    email: str,
+    pw_bcrypt: str,
+    country: str,
+) -> dict[str, Any]:
+    """Create a new player in the database."""
+    query = f"""\
+        INSERT INTO users (name, safe_name, email, pw_bcrypt, country)
```
Comment (cmyui) [inline app/repositories/players.py:51]: priv?

Comment (7ez) [reply app/repositories/players.py:51]: no privileges :trollface:

File: app/api/domains/api.py [hunk 2]
```diff
@@ -168,19 +160,9 @@ async def api_get_player_info(
...
-        )
+        user_info = await players_repo.fetch_one(name=username)
     else:  # if user_id
-        user_info = await app.state.services.database.fetch_one(
-            "SELECT id, name, safe_name, "
-            "priv, clan_id, country, silence_end, donor_end "
-            "FROM users WHERE id = :userid",
-            {"userid": user_id},
-        )
+        user_info = await players_repo.fetch_one(id=user_id)
```
Comment (cmyui) [inline app/api/domains/api.py:165]: this is a behaviour change - the api will now return more data

File: app/repositories/players.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+# | clan_priv         | tinyint(1)    | NO   |     | 0       |                |
+# | preferred_mode    | int           | NO   |     | 0       |                |
+# | play_style        | int           | NO   |     | 0       |                |
+# | custom_badge_name | varchar(16)   | YES  |     | NULL    |                |
+# | custom_badge_icon | varchar(64)   | YES  |     | NULL    |                |
+# | userpage_content  | varchar(2048) | YES  |     | NULL    |                |
+# | api_key           | char(36)      | YES  | UNI | NULL    |                |
+# +-------------------+---------------+------+-----+---------+----------------+
+
+READ_PARAMS = textwrap.dedent(
```
Comment (tsunyoku) [inline app/repositories/players.py:34]: interesting find lol

**Review by tsunyoku — DISMISSED** [review]

**Comment by cmyui** [comment]
2020 me was on some shit with that StartingTimers nullability

File: app/api/domains/api.py (hunk 2 above)
Comment (cmyui) [reply app/api/domains/api.py:165]: confirmed it's not an issue here

File: app/commands.py [hunk 4]
```diff
@@ -2587,7 +2574,7 @@ async def process_commands(
         commands = regular_commands
 
     for cmd in commands:
-        if trigger in cmd.triggers and p.priv & cmd.priv == cmd.priv:
+        if trigger in cmd.triggers and p.priv & cmd.priv != 0:
```
Comment (cmyui) [inline app/commands.py:2577]: hmmm

**Comment by cmyui** [comment]
ugh this pr is getting off topic

File: app/commands.py (hunk 4 above)
Comment (cmyui) [inline app/commands.py:2577]:
```suggestion
        if trigger in cmd.triggers and p.priv & cmd.priv == cmd.priv:
```

File: app/objects/player.py [hunk 5]
```diff
@@ -704,15 +704,13 @@ def leave_match(self) -> None:
             log(f"Match {self.match} finished.")
 
             # cancel any pending start timers
-            if self.match.starting["start"] is not None:
+            if self.match.starting is not None:
                 self.match.starting["start"].cancel()
                 for alert in self.match.starting["alerts"]:
                     alert.cancel()
 
                 # i guess unnecessary but i'm ocd
```
Comment (cmyui) [inline app/objects/player.py:712]:
```suggestion
```

File: app/repositories/players.py [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+
+async def fetch_one(
+    id: Optional[int] = None,
+    name: Optional[str] = None,
+    email: Optional[str] = None,
+    fetch_all_fields: bool = False,  # TODO: probably remove this if possible
+) -> dict[str, Any] | None:
+    """Fetch a single player from the database."""
+    if not (id or name or email):
```
Comment (cmyui) [inline app/repositories/players.py:83]:
```suggestion
    if id is None and name is None and email is None:
```

**Pull request merged**